File: src/problems_all/976_largest_perimeter_triangle/976_largest_perimeter_triangle_v1.go
/*
https://leetcode.com/problems/largest-perimeter-triangle/?envType=study-plan&id=programming-skills-i
https://leetcode.com/problems/largest-perimeter-triangle/

Runtime: 40 ms
Memory: 6.6 MB
*/

package main

import (
	"fmt"
	"sort"
)

func main() {
	nums := []int{1, 2, 1, 10}
	perimeter := largestPerimeter(nums)
	fmt.Println(perimeter)
}

func largestPerimeter(nums []int) int {
	perimeter := 0
	sort.Slice(nums, func(i, j int) bool {
		return nums[i] > nums[j]
	})

	for i := 0; i < len(nums)-2; i++ {
		a, b, c := nums[i], nums[i+1], nums[i+2]

		cond1 := a+b > c
		cond2 := a+c > b
		cond3 := b+c > a
		if cond1 && cond2 && cond3 {
			if a+b+c > perimeter {
				perimeter = a + b + c
				break
			}
		}
	}

	return perimeter
}

File: src/problems_all/976_largest_perimeter_triangle/976_largest_perimeter_triangle_v1.ts
/*
https://leetcode.com/problems/largest-perimeter-triangle/

Runtime: 120 ms
Memory: 46.8 MB
https://leetcode.com/problems/largest-perimeter-triangle/submissions/875605541/
*/

function largestPerimeter(nums: number[]): number {
  let perimeter = 0;
  nums.sort((a, b) => b - a);

  for (let i = 0; i < nums.length - 2; i++) {
    let [a, b, c] = nums.slice(i, i + 3);
    if (b + c > a) {
      perimeter = a + b + c;
      break;
    }
  }

  return perimeter;
}

function main() {
  let nums = [3, 6, 2, 3];
  let result = largestPerimeter(nums);
  console.log(result);
}

main();
